src/mkfs.fat.c

```diff
@@ -1597,6 +1600,15 @@ int main(int argc, char **argv)
 	    }
 	    break;
 
+    case OPT_OFFSET:
+        errno = 0;
+        part_offset = (off_t) strtoll(optarg, &tmp, 0) * sector_size;
```

Comment:
This is incorrect. `sector_size` is set by `-S` parameter, therefore when `--offset` and `-S` are called in specific order result is different. First you need to process all parameters and then do calculations which depends on combination of more parameters.

The simple fix is to replace `part_offset` by `part_sector` and change unit from bytes to sectors. So you would have variables independent of order.